pkg/restapi/issuer/operation/operations.go

```diff
@@ -1016,53 +1038,60 @@ func (o *Operation) validateAndGetConnection(connectData *issuervc.DIDConnectCre
 	return conn, nil
 }
 
-func (o *Operation) createTxn(profile *issuer.ProfileData, state, token string) (string, error) {
-	invitation, err := o.oobClient.CreateInvitation(nil, outofband.WithLabel("issuer"),
-		outofband.WithGoal("", oobGoalCode))
-	if err != nil {
-		return "", fmt.Errorf("failed to create invitation : %w", err)
-	}
-
-	txnID := uuid.New().String()
+func (o *Operation) createTxn( // nolint:funlen
+	profile *issuer.ProfileData,
+	state, token, credScope string,
+) (string, error) {
+	var invBytes []byte
 
-	// store the txn data
-	data := &txnData{
-		IssuerID:          profile.ID,
-		State:             state,
-		DIDCommInvitation: invitation,
-		Token:             token,
-	}
+	if profile.IsDIDCommV2 { // nolint:nestif
+		invitationV2, err := o.oobV2Client.CreateInvitation(
+			outofbandv2.WithFrom(profile.PublicDID),
+			outofbandv2.WithLabel("issuer"),
+			outofbandv2.WithGoal("", oobGoalCode),
+		)
+		if err != nil {
+			return "", fmt.Errorf("failed to create invitation : %w", err)
+		}
 
-	dataBytes, err := json.Marshal(data)
-	if err != nil {
-		return "", fmt.Errorf("failed to marshal txn data: %w", err)
-	}
+		invBytes, err = json.Marshal(invitationV2)
+		if err != nil {
+			return "", fmt.Errorf("marshal invitation: %w", err)
+		}
+	} else {
+		invitationV1, err := o.oobClient.CreateInvitation(nil, outofband.WithLabel("issuer"),
+			outofband.WithGoal("", oobGoalCode))
+		if err != nil {
+			return "", fmt.Errorf("failed to create invitation : %w", err)
+		}
 
-	err = o.txnStore.Put(txnID, dataBytes)
-	if err != nil {
-		return "", fmt.Errorf("failed to store txn data: %w", err)
+		invBytes, err = json.Marshal(invitationV1)
+		if err != nil {
+			return "", fmt.Errorf("marshal invitation: %w", err)
+		}
 	}
 
-	return txnID, nil
-}
+	invitation := &wallet.GenericInvitation{}
 
-func (o *Operation) createTxnWithCredScope(profile *issuer.ProfileData, credScope string) (string, error) {
-	invitation, err := o.oobClient.CreateInvitation(nil, outofband.WithLabel("issuer"),
-		outofband.WithGoal("", oobGoalCode))
+	err := json.Unmarshal(invBytes, invitation)
 	if err != nil {
-		return "", fmt.Errorf("failed to create invitation : %w", err)
+		return "", fmt.Errorf("unmarshal invitation: %w", err)
 	}
 
 	txnID := uuid.New().String()
 
-	token := uuid.New().String()
+	if credScope != "" {
```

Comment:
can credential scope be empty?

Reply:
@sudeshrshetty credScope is empty when the adapter is doing the state flow rather than the cred scope flow.